go/inst/instance_topology.go

```diff
@@ -1675,6 +1677,33 @@ func TakeSiblings(instanceKey *InstanceKey) (instance *Instance, takenSiblings i
 	return instance, len(relocatedReplicas), err
 }
 
+// Created this function to allow a hook to be called after a successful TakeMaster event
+func TakeMasterHook(master *Instance, slave *Instance) {
+	masterInstance := master.Key
+	slaveInstance := slave.Key
+	env := goos.Environ()
+	env = append(env, fmt.Sprintf("ORC_SUCCESSOR_HOST=%s", masterInstance))
+	env = append(env, fmt.Sprintf("ORC_FAILED_HOST=%s", slaveInstance))
+	successor := fmt.Sprintf("%s", masterInstance)
+	oldmaster := fmt.Sprintf("%s", slaveInstance)
+	if config.Config.PostTakeMasterProcesses != nil {
+		processCount := len(config.Config.PostTakeMasterProcesses)
+		for i, command := range config.Config.PostTakeMasterProcesses {
+			fullDescription := fmt.Sprintf("PostTakeMasterProcesses hook %d of %d", i+1, processCount)
+			log.Debugf("Take-Master: PostTakeMasterProcesses: Calling %+s", fullDescription)
+			start := time.Now()
+			if err := os.CommandRun(command, env, successor, oldmaster); err == nil {
+				info := fmt.Sprintf("Completed %s in %v", fullDescription, time.Since(start))
+				log.Infof("Take-Master: %s", info)
+			} else {
+				info := fmt.Sprintf("Execution of PostTakeMasterProcesses failed in %v with error: %v", time.Since(start), err)
+				log.Infof("Take-Master: %s", info)
```

Comment:
Remove that line. If we have `Errorf` we don't want to have same `Infof`.
```suggestion
```

Reply:
Removed as suggested, thanks Shlomi.

Reply:
I've pushed the changes, please let me know if this is ok. Thanks for all the help and input 😄  👍